design/build-clusters.md

```diff
@@ -0,0 +1,140 @@
+# Multiple Build Cluster 
+
+## Context
+
+Build a scalable Screwdriver build infrastructure. 
+
+## Objective 
+
+1. Running builds in parallel with help of multiple build clusters.
+2. Users can bring in their own build clusters for their own specific infrastructure needs.
+3. Support build cluster in AWS
+
+## Proposal
+
+1. Implicit build clusters
+	Screwdriver maintains its own multiple build cluster infrastructure.
+
+2. Explicit build clusters
+	Users can bring in their own build cluster infrastructure. 
+
+As part of cluster onboarding process for above both options cluster admin should register their build cluster information with Screwdriver. Below details are required to register a build cluster.
+	1. Cluster name
+	2. Status - cluster status
+	3. Queue name
+	4. Authorization (initial phase will support JWT with key and sign options)
+	5. SCM Organization
```

Comment:
let's elaborate there to explain what is this for. Based on @jithin1987, this will be used to determine if a job has permission to run on a certain build cluster. e.g. only pipelines under mobile org can run in IOS cluster